zokrates_core/src/static_analysis/constant_inliner.rs

```diff
@@ -165,9 +172,8 @@ impl<'ast, T: Field> ResultFolder<'ast, T> for ConstantInliner<'ast, T> {
                         self.fold_function_symbol(fun)?,
                     ))
                 })
-                .collect::<Result<Vec<_>, _>>()
+                .collect::<Result<Vec<_>, _>>()?
```

Comment:
This was actually pretty bad, forgot to call "?" and then called `into_iter` on an `Err` which yields an empty iterator, returning an empty array eventually..